nevergrad/functions/ttp/core.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+import numpy as np
+import nevergrad as ng
+from nevergrad.parametrization import parameter as p
+from .. import base
+
+
+class TTPSolution:
+    def __init__(self, tspTour, packingPlan):
+
+        self.tspTour = tspTour
+        self.packingPlan = packingPlan
+
+        self.fp = float("-inf")
+        self.ft = float("inf")
+        self.ftraw = math.inf
+        self.ob = float("-inf")
+        self.wend = float("inf")
+        self.wendUsed = float("inf")
+        self.computationTime = math.inf
+
+    def reset(self):
+
+        self.fp = float("-inf")
+        self.ft = float("inf")
+        self.ftraw = math.inf
+        self.ob = float("-inf")
+
+        self.wend = float("inf")
+        self.wendUsed = float("inf")
+        self.computationTime = math.inf
+
+
+class TTPInstance(base.ExperimentFunction):
+    def __init__(self, file: str = "./instances/a20_n95_01.ttp") -> None:

```

Comment:
depending on how you want to use this type of files, this may not be the best option as parameter.
You could either give them a number each, it's simpler to use but then you can't use external files directly.
Or you could provide just the name, and then if you want a new file you add it to the folder.

Currently, I think the default path can't work, because the working directory is not the directory which contains the file, and you should not expect it is. This is also what triggers the continuous integration error in the PR.

A first step to check this would actually be to create a unit test and make it work.
That could/should be something as simple as this: https://github.com/facebookresearch/nevergrad/blob/edac4eaa65db5e5e01dc0f1ebcc5134c47b78060/nevergrad/functions/powersystems/test_core.py#L12-L18

Reply:
pytest nevergrad/functions/powersystems/test_core.py   will launch the test in that file.
You can réplication this file in the directory nevergrad/functions/ttt/test_core.py